### PR TITLE
Feature/bimdata menu components

### PR DESCRIPTION
**Pull request by LrxGaelle**
- [ ] I have tested my component
- [ ] I have updated the documentation or there is no documentation needed

### CODE REVIEW

**Review by NicolasRichel — APPROVED** [review]

File: src/BIMDataComponents/BIMDataMenuInline/BIMDataMenuInline.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  </div>
+</template>
+
+<script>
+import clickaway from "../../BIMDataDirectives/click-away.js";
+import BIMDataButton from "../BIMDataButton/BIMDataButton.vue";
+import BIMDataIcon from "../BIMDataIcon/BIMDataIcon.vue";
+export default {
+  directives: { clickaway },
+
```
Comment (Kurtil) [inline src/BIMDataComponents/BIMDataMenuInline/BIMDataMenuInline.vue:49]: What is this space for ?

File: src/web/views/Components/Menu/BasicMenu.vue [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            &lt;/template&gt;
+          &lt;/BIMDataMenu&gt;
+        </pre>
+      </template>
+    </ComponentCode>
+  </div>
+</template>
+
+<script>
+import { basicOptions } from "./option-sets";
```
Comment (Kurtil) [inline src/web/views/Components/Menu/BasicMenu.vue:50]: Missing file extension.

File: src/web/views/Components/Menu/slots-menu-inline.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default [
+  ["Slot name", "Description"],
+  ["button", "Use this slot for custom button icon to open the menu"],
+  ["submenu", "Use this slot for add a submenu"],
```
Comment (Kurtil) [inline src/web/views/Components/Menu/slots-menu-inline.js:4]: Use this slot *to* add a submenu

File: src/web/views/Components/Menu/Menus.vue [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import BIMDataTable from "../../../../BIMDataComponents/BIMDataTable/BIMDataTable.vue";
+import BIMDataText from "../../../../BIMDataComponents/BIMDataText/BIMDataText.vue";
+
+import MenuInline from "./MenuInline.vue";
+import BasicMenu from "./BasicMenu.vue";
+
+import propsMenuInline from "./props-menu-inline";
+import slotsMenuInline from "./slots-menu-inline";
+import propsBasicMenu from "./props-basic-menu";
+import slotsBasicMenu from "./slots-basic-menu";
```
Comment (Kurtil) [inline src/web/views/Components/Menu/Menus.vue:61]: Missing file extensions.

**Review by Kurtil — CHANGES_REQUESTED** [review]

File: src/BIMDataComponents/BIMDataMenu/BIMDataMenu.vue [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        currentChild == undefined ||
+        currentChild.length < 1
+      ) {
+        return;
+      }
+
+      const itemPos = currentItem[0].getBoundingClientRect();
+      const childPos = currentChild[0].getBoundingClientRect();
+
+      return (childPos.height - itemPos.height) * -1;
```
Comment (Kurtil) [inline src/BIMDataComponents/BIMDataMenu/BIMDataMenu.vue:128]: You can use destructuring here if you want:

```js
const { height: itemPosHeight } = currentItem[0].getBoundingClientRect();
const { height: childPosHeight } = currentChild[0].getBoundingClientRect();

return (childPosHeight - itemPosHeight) * -1;
```

File: src/BIMDataComponents/BIMDataMenu/BIMDataMenu.vue [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  data() {
+    return {
+      isItemHover: false,
+      currentItemText: null,
+      displayed: false,
+    };
+  },
+  methods: {
+    definePos(item) {
+      if (!item.children.position || !item.children.position === "up") return 0;
```
Comment (Kurtil) [inline src/BIMDataComponents/BIMDataMenu/BIMDataMenu.vue:111]:
```js
!item.children.position === "up"
```

may be replace by:

```js
item.children.position !== "up"
```

File: src/web/views/Components/Menu/props-menu-inline.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable */
+export default [
+  [
+    "Name", "Type", "Default value", "Description"
+  ],
+  [
+    "isSubmenuRight",
+    "Boolean",
+    "true",
+    "Set this props to 'false' if you want to open the submenu on the left",
```
Comment (Kurtil) [inline src/web/views/Components/Menu/props-menu-inline.js:10]: Remove 'if you want'

File: src/web/views/Components/Menu/props-menu-inline.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    "width",
+    "String",
+    "200px",
+    "Use this props to custom width of BIMDataMenuInline component.",
+  ],
+  [
+    "isClickAway",
+    "Boolean",
+    "true",
+    "Set this props to false if you don't want the menu component to close when you click outside.",
```
Comment (Kurtil) [inline src/web/views/Components/Menu/props-menu-inline.js:22]: "If false, the menu component closes when you click outside."

File: src/web/views/Components/Menu/props-basic-menu.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    "width",
+    "String",
+    "200px",
+    "Use this props to custom width of BIMDataMenu component.",
+  ],
+  [
+    "isClickAway",
+    "Boolean",
+    "true",
+    "Set this props to false if you don't want the menu component to close when you click outside.",
```
Comment (Kurtil) [inline src/web/views/Components/Menu/props-basic-menu.js:34]: "If false, the menu component closes when you click outside."

**Review by Kurtil — CHANGES_REQUESTED** [review]

**Review by Kurtil — APPROVED** [review]

**Pull request merged**